src/display/lv_display_private.h

```diff
@@ -99,6 +99,8 @@ struct _lv_display_t {
     /** Double buffer sync areas (redrawn during last refresh) */
     lv_ll_t sync_areas;
 
+    lv_draw_buf_t _static_buf1; /*Used when user pass in a raw buffer as display draw buffer*/
```

Comment:
Can't we have only `lv_draw_buf_t buf_1/2` (not pointer) and copy the draw_bufs by value in `lv_display_set_draw_buffers`?
```c
void lv_display_set_draw_buffers(lv_display_t * disp, lv_draw_buf_t * buf1, lv_draw_buf_t * buf2)
{
    if(disp == NULL) disp = lv_display_get_default();
    if(disp == NULL) return;

    disp->buf_1 = *buf1; //copy by value
    disp->buf_2 = *buf2;
    disp->buf_act = &disp->buf_1;
}
```

This way the `draw_buf` parameters don't have to be `static`.

Reply:
This comes to the topic we discuss life cycle of lv_draw_buf_t. We should never copy by struct. For example, memory leak when buf1 = lv_draw_buf_create();

Also It's a minimal change not break existing logic.

Reply:
I see, so when the display is destroyed, you won't be able the free the `draw_buf`s. That's true.